pkg/oam/util/test_utils.go

```diff
@@ -0,0 +1,90 @@
+package util
```

Comment:
we can move it to separate package outside of `pkg` to increase the test coverage

Reply:
I am adding tests right now. See latest commit.